platform/android/llapp_main.lua

```diff
@@ -15,6 +15,72 @@ end
 -- path to primary external storage partition
 local path = android.getExternalStoragePath()
 
+-- execute user scripts before patch.lua
+local lfs = require("libs/libkoreader-lfs")
+
+local user_dir = android.getExternalStoragePath() .. "/koreader"
+local afterupdate_marker = android.dir .. "/afterupdate.marker"
+local run_once_scripts = user_dir .. "/scripts.afterupdate"
+local run_always_scripts = user_dir .. "/scripts.always"
+
+local shell_scripts = {}
+
+local function getUserScripts(path)
+    shell_scripts = {} -- zero list, as getUserScripts could be called multiple times
+    local ret = lfs.attributes(path)
+    if ret == nil then
+        android.LOGI("no script folder " .. path .. " found!")
```

Comment:
There's no need for this information in 99'99999% of the cases.

Reply:
You are right, but in case something goes wrong, if is handy to see (and not to forget) that there are no user scripts involved.